source/python/cell_accessor_wrapper.cc

```diff
@@ -0,0 +1,472 @@
+      {
+        dealii::CellAccessor<2,2> *other_cell =
+          static_cast<dealii::CellAccessor<2,2>*>(other.cell_accessor);
+        cell_accessor = new dealii::CellAccessor<2,2>(*other_cell);
+      }
+    else if ((dim == 2) && (spacedim == 3))
+      {
+        dealii::CellAccessor<2,3> *other_cell =
+          static_cast<dealii::CellAccessor<2,3>*>(other.cell_accessor);
+        cell_accessor = new dealii::CellAccessor<2,3>(*other_cell);
+      }
+    else if ((dim == 3) && (spacedim == 3))
+      {
+        dealii::CellAccessor<3,3> *other_cell =
+          static_cast<dealii::CellAccessor<3,3>*>(other.cell_accessor);
+        cell_accessor = new dealii::CellAccessor<3,3>(*other_cell);
+      }
+    else
+      AssertThrow(false, dealii::ExcMessage("Wrong dim-spacedim combination."));
+  }
+
+
+
+
+  CellAccessorWrapper::CellAccessorWrapper(TriangulationWrapper &triangulation_wrapper,
+                                           const int             level,
+                                           const int             index)
+  {
+    dim = triangulation_wrapper.get_dim();
+    spacedim = triangulation_wrapper.get_spacedim();
+    if ((dim == 2) && (spacedim == 2))
+      {
+        dealii::Triangulation<2,2> *tmp = static_cast<dealii::Triangulation<2,2>*> (
+                                            triangulation_wrapper.get_triangulation());
+        cell_accessor = new dealii::CellAccessor<2,2>(tmp ,level, index);
+      }
+    else if ((dim == 2) && (spacedim == 3))
+      {
+        dealii::Triangulation<2,3> *tmp = static_cast<dealii::Triangulation<2,3>*> (
+                                            triangulation_wrapper.get_triangulation());
+        cell_accessor = new dealii::CellAccessor<2,3>(tmp, level, index);
+      }
+    else if ((dim == 3) && (spacedim == 3))
+      {
+        dealii::Triangulation<3,3> *tmp = static_cast<dealii::Triangulation<3,3>*> (
+                                            triangulation_wrapper.get_triangulation());
+        cell_accessor = new dealii::CellAccessor<3,3>(tmp, level, index);
+      }
+    else
+      AssertThrow(false, dealii::ExcMessage("Wrong dim-spacedim combination."));
+  }
+
+
+
+  CellAccessorWrapper::~CellAccessorWrapper()
+  {
+    if (dim != -1)
+      {
+        if ((dim == 2) && (spacedim == 2))
+          {
+            // We cannot call delete on a void pointer so cast the void pointer back
+            // first.
+            dealii::CellAccessor<2,2> *tmp =
+              static_cast<dealii::CellAccessor<2,2>*>(cell_accessor);
+            delete tmp;
+          }
+        else if ((dim == 2) && (spacedim == 3))
+          {
+            dealii::CellAccessor<3,3> *tmp =
+              static_cast<dealii::CellAccessor<3,3>*>(cell_accessor);
+            delete tmp;
+          }
+        else
+          {
+            dealii::CellAccessor<3,3> *tmp =
+              static_cast<dealii::CellAccessor<3,3>*>(cell_accessor);
+            delete tmp;
```

Comment:
I only use ```AssertThrow``` in the constructors to make sure the object is built with valid ```dim``` and ```spacedim``` values. Once the object is created, the values cannot be changed and there is no point in checking them anymore.